plugins/tpm/fu-self-test.c

```diff
@@ -26,7 +26,8 @@ fu_tpm_device_1_2_func(void)
 	g_autoptr(FuContext) ctx = fu_context_new();
 	g_autoptr(FuPlugin) plugin = fu_plugin_new(ctx);
 	g_autoptr(FuSecurityAttrs) attrs = fu_security_attrs_new();
-	g_autoptr(FwupdSecurityAttr) attr = NULL;
+	g_autoptr(FwupdSecurityAttr) attr0 = NULL;
```

Comment:
I'm not sure if I introduced this or not, but I definitely could have since I'm used to C++'s `unique_ptr` which does the right thing when re-assigning. In hindsight, it's obvious why the same is not possible in C.

Is there some systematic way to catch this, eg. by running the tests under LeakSanitizer or Valgrind in CI?

Reply:
> Is there some systematic way to catch this

I do runs with Valgrind, Coverity and `-Db_sanitize=none, address, thread, undefined, memory, address,undefined` before each release but there are too many false positives to do this in CI. Valgrind gets closest, but it's really really slow.